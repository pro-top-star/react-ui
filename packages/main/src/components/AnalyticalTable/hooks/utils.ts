export const makeTemplateColumns = (columns, resizedColumns) => {
  return columns
    .map((column) => {
      if (resizedColumns[column.id]) {
        return `${resizedColumns[column.id]}px`;
      }
      return `minmax(${column.minWidth}px, ${column.width || '1fr'})`;
    })
    .join(' ');
};
