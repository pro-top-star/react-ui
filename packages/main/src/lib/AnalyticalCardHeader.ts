import { AnalyticalCardHeader } from '../components/AnalyticalCard/header/AnalyticalCardHeader';

export { AnalyticalCardHeader };
