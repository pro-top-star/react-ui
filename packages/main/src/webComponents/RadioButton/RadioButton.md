[Link to UI5 Web Components Playground](https://sap.github.io/ui5-webcomponents/playground/components/RadioButton)

```jsx
import { RadioButton } from '@ui5/webcomponents-react/lib/RadioButton';
```
