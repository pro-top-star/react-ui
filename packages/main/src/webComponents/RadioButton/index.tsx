import { Event } from '@ui5/webcomponents-react-base/lib/Event';
import { ValueState } from '@ui5/webcomponents-react/lib/ValueState';
import { withWebComponent } from '@ui5/webcomponents-react/lib/withWebComponent';
import UI5RadioButton from '@ui5/webcomponents/dist/RadioButton';
import React, { FC } from 'react';
import { WithWebComponentPropTypes } from '../../internal/withWebComponent';

export interface RadioButtonPropTypes extends WithWebComponentPropTypes {
  disabled?: boolean; // @generated
  readonly?: boolean; // @generated
  selected?: boolean; // @generated
  text?: string; // @generated
  valueState?: ValueState; // @generated
  name?: string; // @generated
  value?: string; // @generated
  onSelect?: (event: Event) => void; // @generated
}

/**
 * In order to use the Radio Button in Compact Size, please add the following snippet to your `index.html`:
 * ```
 * <script data-ui5-config type="application/json">
 *  { "compactSize": true }
 * </script>
 * ```
 *
 * This Storybook is currently only working in ContentDensity=Cozy.
 */
const RadioButton: FC<RadioButtonPropTypes> = withWebComponent<RadioButtonPropTypes>(UI5RadioButton);

RadioButton.displayName = 'RadioButton';

RadioButton.defaultProps = {
  valueState: ValueState.None // @generated
};

export { RadioButton };
