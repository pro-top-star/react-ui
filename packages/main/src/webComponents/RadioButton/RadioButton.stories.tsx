import { createSelectArgTypes } from '@shared/stories/createSelectArgTypes';
import { RadioButton } from '@ui5/webcomponents-react/lib/RadioButton';
import { ValueState } from '@ui5/webcomponents-react/lib/ValueState';
import React from 'react';

export default {
  title: 'UI5 Web Components / RadioButton',
  component: RadioButton,
  argTypes: {
    ...createSelectArgTypes({ value: ValueState }),
    ref: {
      type: null
    }
  },
  args: {
    text: 'Radio Button Text',
    value: ValueState.None,
    valueState: 'None'
  }
};

export const generatedDefaultStory = (props) => (
  <RadioButton
    disabled={props.disabled}
    name={props.name}
    readonly={props.readonly}
    selected={props.selected}
    text={props.text}
    value={props.value}
    valueState={props.valueState}
    wrap={props.wrap}
    onSelect={props.onSelect}
  />
);

generatedDefaultStory.storyName = 'Generated default story';
