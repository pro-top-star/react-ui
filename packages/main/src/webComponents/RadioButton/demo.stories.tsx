import { action } from '@storybook/addon-actions';
import { boolean, select, text } from '@storybook/addon-knobs';
import { RadioButton } from '@ui5/webcomponents-react/lib/RadioButton';
import { ValueState } from '@ui5/webcomponents-react/lib/ValueState';
import React from 'react';
import notes from './RadioButton.md';

export default {
  title: 'UI5 Web Components / RadioButton',
  component: RadioButton,
  parameters: { notes }
};

export const defaultStory = () => (
  <RadioButton
    disabled={boolean('disabled', false)}
    readonly={boolean('readonly', false)}
    selected={boolean('selected', false)}
    text={text('text', 'Option A')}
    valueState={select('valueState', ValueState, null)}
    onSelect={action('onSelect')}
  />
);

defaultStory.story = {
  name: 'Default'
};
