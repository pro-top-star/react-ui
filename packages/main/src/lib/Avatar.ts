import { Avatar } from '../webComponents/Avatar';

export { Avatar };
