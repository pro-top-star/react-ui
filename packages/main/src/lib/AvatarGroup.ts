import { AvatarGroup } from '../webComponents/AvatarGroup';
import type { AvatarGroupPropTypes } from '../webComponents/AvatarGroup';

export { AvatarGroup };
export type { AvatarGroupPropTypes };
