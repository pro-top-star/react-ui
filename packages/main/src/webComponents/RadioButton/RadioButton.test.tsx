import { mountThemedComponent } from '@shared/tests/utils';
import { RadioButton } from '@ui5/webcomponents-react/lib/RadioButton';
import React from 'react';

describe('RadioButton', () => {
  test('Basic Test (generated)', () => {
    const wrapper = mountThemedComponent(<RadioButton />);
    expect(wrapper.render()).toMatchSnapshot();
  });
});
