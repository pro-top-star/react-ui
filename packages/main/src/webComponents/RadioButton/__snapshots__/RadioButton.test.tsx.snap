// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioButton Basic Test (generated) 1`] = `
<ui5-radiobutton
  value-state="None"
/>
`;
